README.md

```diff
@@ -18,11 +18,73 @@ OSConfig's C/C++ code currently targets compliance with C11.
 
 ### Prerequisites
 
-Make sure all dependencies are installed. For example, on Ubuntu:
+Build environments have many dependencies required, the easiest way to get started is to use our pre-defined container environments in [devops/docker](devops/docker/).
+
+Make sure all dependencies are installed for your distribution. All of our supported distributions are documented in the Dockerfiles under [devops/docker](devops/docker/) (additional packages may be present for CI which are not necessary for building). The following packages are typically required however the package names may vary across distributions. There might also be packages that are out-of-date so they are built installed from source.
+
+**Common dependencies across distributions:**
+- git
+- cmake (>= 3.16.0)
+- build-essential (gcc >= 4.4.7, g++, make)
+- curl
+- autoconf
+- perl (perl-core, perl-IPC-Cmd)
+- tar
+- unzip
+- zip
+- python3
+- gtest/gmock
+
+Refer to the specific Dockerfile for your distribution under [devops/docker/](devops/docker/) for the complete and up-to-date list of dependencies. See the following example.
+
+#### Example prerequisite install on Ubuntu 24.04
+Using the pre-defined [devops/docker/ubuntu-24.04-amd64/Dockerfile](devops/docker/ubuntu-24.04-amd64/Dockerfile) we can simply use the `RUN` commands to install all needed pre-requisites. We also need to run `sudo -i` since many commands require `root`.
 
 ```bash
-sudo apt-get update
-sudo apt-get install -y git cmake build-essential curl libcurl4-openssl-dev libssl-dev uuid-dev libgtest-dev libgmock-dev rapidjson-dev
+# login as `root` so we don't have to `sudo` all the time
+sudo -i
+
+apt -y update && apt-get -y install software-properties-common
+apt -y update && apt-get -y install \
+    apt-transport-https \
+    bc \
+    build-essential \
+    clang \
+    clang-tools \
+    cmake \
+    curl \
+    file \
+    gcovr\
+    git \
+    jq \
+    libasan8 \
+    libcurl4-openssl-dev \
+    libgmock-dev \
+    libgtest-dev \
+    libssl-dev \
+    libubsan1 \
+    ninja-build \
+    pkg-config \
+    python3-jsonschema \
+    rapidjson-dev \
```

Comment:
>rapidjson-dev [](http://example.com/codeflow?start=4&length=13)

Why rapidjson-dev when we already had rapidjson pulled into the project and in process to replace it with parson?

Remember, rapidjson is not an approved parser/serializer.